x-pack/plugins/cases/public/components/user_actions/index.tsx

```diff
@@ -35,24 +38,27 @@ export const UserActions = React.memo((props: UserActionTreeProps) => {
     userActionsStats,
   } = props;
   const { detailName: caseId } = useCaseViewParams();
+  const {
+    isLoadingInfiniteUserActions,
+    lastPage,
+    caseUserActions: infiniteUserActions,
+    hasNextPage,
+    fetchNextPage,
+    showLoadMore,
+    showBottomList,
+  } = useUserActionsPagination({
+    userActivityQueryParams,
+    userActionsStats,
+    caseId: caseData.id,
+    isExpandable: true,
+  });
 
-  const lastPage = useMemo(() => {
-    if (!userActionsStats) {
-      return 1;
-    }
-
-    const perPage = userActivityQueryParams.perPage;
-
-    switch (userActivityQueryParams.type) {
-      case 'action':
-        return Math.ceil(userActionsStats.totalOtherActions / perPage);
-      case 'user':
-        return Math.ceil(userActionsStats.totalComments / perPage);
-      case 'all':
-      default:
-        return Math.ceil(userActionsStats.total / perPage);
-    }
-  }, [userActionsStats, userActivityQueryParams]);
+  const { caseUserActions, isLoadingUserActions } = useUserActionsPagination({
+    userActivityQueryParams: { ...userActivityQueryParams, page: lastPage },
```

Comment:
I don't think we need two calls of the `useUserActionsPagination`. We can combine the needed logic into one call. Also by doing two calls of the `useUserActionsPagination` call we also do four calls to the backend two for each invocation.

Reply:
Good point!